prometheus-ksonnet/lib/prometheus-config.libsonnet

```diff
@@ -19,387 +20,44 @@ local prometheus = import 'prometheus/prometheus.libsonnet';
-        kubernetes_sd_configs: [{
-          role: 'node',
-        }],
-
-        // Couldn't get prometheus to validate the kublet cert for scraping, so don't bother for now
-        tls_config: {
-          ca_file: '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
-          insecure_skip_verify: $._config.prometheus_insecure_skip_verify,
-        },
-        bearer_token_file: '/var/run/secrets/kubernetes.io/serviceaccount/token',
-
-        relabel_configs: [
-          {
-            target_label: '__address__',
-            replacement: $._config.prometheus_api_server_address,
-          },
-          {
-            target_label: '__scheme__',
-            replacement: 'https',
-          },
-          {
-            source_labels: ['__meta_kubernetes_node_name'],
-            regex: '(.+)',
-            target_label: '__metrics_path__',
-            replacement: '/api/v1/nodes/${1}/proxy/metrics',
-          },
-        ],
-      },
-
-      // As of k8s 1.7.3, cAdvisor metrics are available via kubelet using the /metrics/cadvisor path
-      {
-        job_name: 'kube-system/cadvisor',
-        kubernetes_sd_configs: [{
-          role: 'node',
-        }],
-        scheme: 'https',
-
-        bearer_token_file: '/var/run/secrets/kubernetes.io/serviceaccount/token',
-        tls_config: {
-          ca_file: '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
-          insecure_skip_verify: $._config.prometheus_insecure_skip_verify,
-        },
-
-        relabel_configs: [
-          {
-            target_label: '__address__',
-            replacement: $._config.prometheus_api_server_address,
-          },
-          {
-            source_labels: ['__meta_kubernetes_node_name'],
-            regex: '(.+)',
-            target_label: '__metrics_path__',
-            replacement: '/api/v1/nodes/${1}/proxy/metrics/cadvisor',
-          },
-        ],
-
-        metric_relabel_configs: [
-          // Drop container_* metrics with no image.
-          {
-            source_labels: ['__name__', 'image'],
-            regex: 'container_([a-z_]+);',
-            action: 'drop',
-          },
-
-          // Drop a bunch of metrics which are disabled but still sent, see
-          // https://github.com/google/cadvisor/issues/1925.
-          {
-            source_labels: ['__name__'],
-            regex: 'container_(network_tcp_usage_total|network_udp_usage_total|tasks_state|cpu_load_average_10s)',
-            action: 'drop',
-          },
-        ],
-      },
+      scrape_configs.kubelet($._config.prometheus_api_server_address)
+      + (
+        // Couldn't get prometheus to validate the kublet cert for scraping, so
+        // don't bother for now
+        if $._config.prometheus_insecure_skip_verify
+        then scrape_configs.insecureSkipVerify
+        else {}
+      ),
+
+      // As of k8s 1.7.3, cAdvisor metrics are available via kubelet using the
+      // /metrics/cadvisor path
```

Comment:
End with period. But perhaps we can also remove the whole comment as K8s 1.7 is ages ago.